src/push.js

```diff
@@ -64,28 +64,47 @@ class Push extends EventEmitter {
             + ' setting up your project.');
         }
 
+        return new Promise((resolve) => {
+          if (phonegapPush) {
+            return phonegapPush.unregister(() => {
+              resolve();
+            }, () => {
+              resolve();
+            });
+          }
+
+          return resolve();
+        });
+      })
+      .then(() => {
         return new Promise((resolve, reject) => {
-          this.phonegapPush = global.PushNotification.init(options);
+          phonegapPush = global.PushNotification.init(options);
```

Comment:
@thomasconner wouldn't this replace an existing instance of `phonegapPush` with a new instance? Is that what is intended?